HIP/hip-850.md

```diff
@@ -0,0 +1,70 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Account
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2024-04-03T07:00:00Z
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks, TierBot
+updated: 2024-03-20
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury account. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury account, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
```

Comment:
Can you expand on what the limitation is with some use cases where community might be blocked or find it challenging?

Reply:
Currently it's impossible to update the metadata of any NFT. After HIP-657 goes into production, this will be more accessible but still doesn't lend to this specific use case of immutable outside the treasury wallet, and mutable in the treasury wallet.

Example use cases
- A game piece that can be updated when you stake the piece and play the game, but cannot be adapted when it's outside the treasury wallet, guaranteeing that the metadata isn't adapted to increase or decrease the secondary market value remotely

This cannot be achieved while maintaining provenance of history of changes on one serial number of a token id.

Reply:
> After HIP-657 goes into production, this will be more accessible but still doesn't lend to this specific use case of immutable outside the treasury wallet, and mutable in the treasury wallet.

After making this comment I came back to the above. It seems like the above is the main aim of the HIP right and the mechanism being used is the supply key? HIP 657 create mutability everywhere when a metadata_key is present but for some scenarios you'd like to have pseudo-dynamic NFTs (mutable in treasury balance and immutable outside)

Reply:
@ty-swirldslabs So this is specifically when the token doesn't have an admin key or metadata key. We will be able to update metadata using supply key _only_ when in treasury account.